api/mod/src/database/db_models.py

```diff
@@ -0,0 +1,77 @@
+"""
+Defines the database models.
+"""
+
+import hashlib
+import hmac
+import os
+from typing import Optional
+
+from sqlalchemy import Column, Integer, String
+from sqlalchemy.orm import Session
+
+from mod.src.models.user import UserCreate
+
+from .database import Base
+
+
+class User(Base):
+    """The basic user database model"""
+
+    __tablename__ = "users"
+
+    id: int = Column(Integer, primary_key=True, index=True)
+    username: str = Column(String, unique=True, index=True)
+    email: str = Column(String, unique=True, index=True)
+    hashed_password: bytes = Column(String)
+    salt: bytes = Column(String)
+
+    @classmethod
+    def with_password(cls, username: str, email: str, password: str) -> "User":
```

Comment:
So this is a problem in python, that a classmethod cant return its class (from type hints perspective) because the class is not fully defined yet. So one solution is to wrap the type hint into a String, or use the `from future import annotations` thing. Which one would we prefer? (I just remember that we have the annotation import in a different place already, so we could make it consistent)

Reply:
Yes, let's make consistent. I'd prefer this variant to the future thingy, what do you think?